pwiz_tools/Skyline/TestPerf/CarafeBuildLibraryTest.cs

```diff
@@ -0,0 +1,836 @@
+﻿/*
+ * Author: David Shteynberg <dshteyn .at. proteinms.net>,
+ *                  MacCoss Lab, Department of Genome Sciences, UW
+ *
+ * Copyright 2025 University of Washington - Seattle, WA
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using pwiz.Common.SystemUtil;
+using pwiz.Skyline;
+using pwiz.Skyline.Alerts;
+using pwiz.Skyline.Controls;
+using pwiz.Skyline.Controls.Graphs;
+using pwiz.Skyline.Model;
+using pwiz.Skyline.Model.Irt;
+using pwiz.Skyline.Model.Lib;
+using pwiz.Skyline.Model.Lib.Carafe;
+using pwiz.Skyline.Model.Tools;
+using pwiz.Skyline.Properties;
+using pwiz.Skyline.SettingsUI;
+using pwiz.Skyline.SettingsUI.Irt;
+using pwiz.Skyline.ToolsUI;
+using pwiz.Skyline.Util;
+using pwiz.SkylineTestUtil;
+
+namespace TestPerf
+{
+    [TestClass]
+    public class CarafeBuildLibraryTest : AbstractFunctionalTestEx
+    {
+
+        /// <summary>
+        /// When true Python installation is forced by deleting any old installation
+        /// </summary>
+        private bool IsCleanPythonMode => true;
+
+        private bool RunExtendedTest => true;
+
+        /// <summary>
+        /// When true console output is added to clarify what the test has accomplished
+        /// </summary>
+        public bool IsVerboseMode => false;
+
+
+        /// <summary>
+        /// When true the test write the Python hash value for <see cref="Settings.PythonEmbeddableHash"/>
+        /// </summary>
+        protected override bool IsRecordMode => false;
+
+        public string LogOutput => TestContext.GetTestResultsPath("TestConsole.log");
+
+        public IAsynchronousDownloadClient TestDownloadClient { get; set; }
+        private const string TESTDATA_URL = @"https://skyline.ms/tutorials/CarafeTestData/";
```

Comment:
This is not typically how we download perftest data. Is there a reason for this (perhaps you just didn't know how to it's typically done)?